pyqtgraph/graphicsItems/GridItem.py

```diff
@@ -1,33 +1,99 @@
+from .UIGraphicsItem import UIGraphicsItem
 import numpy as np
 from ..Point import Point
 from .. import functions as fn
+from .. import getConfigOption
 
 __all__ = ['GridItem']
 class GridItem(UIGraphicsItem):
     """
     **Bases:** :class:`UIGraphicsItem <pyqtgraph.UIGraphicsItem>`
-    
+
     Displays a rectangular grid of lines indicating major divisions within a coordinate system.
     Automatically determines what divisions to use.
     """
-    
-    def __init__(self):
+
+    def __init__(self, pen='default', textColor='default'):
         UIGraphicsItem.__init__(self)
         #QtGui.QGraphicsItem.__init__(self, *args)
         #self.setFlag(QtGui.QGraphicsItem.ItemClipsToShape)
         #self.setCacheMode(QtGui.QGraphicsItem.DeviceCoordinateCache)
-        
+
+        self.opts = {}
+
+        if pen == 'default':
+            self.setPen()
+        else:
+            self.setPen(pen)
+
+        if textColor == 'default':
+            self.setTextColor()
+        else:
+            self.setTextColor(textColor)
+
+        self.setTickSpacing(x=[None, None, None], y=[None, None, None])
+
+
+    def setPen(self, *args, **kwargs):
+        """Set the pen used to draw the grid."""
+        if args or kwargs:
+            self.opts['pen'] = fn.mkPen(*args, **kwargs)
+        else:
+            self.opts['pen'] = fn.mkPen(getConfigOption('foreground'))
+
+        self.picture = None
+        self.update()
+
+    def setTextColor(self, *args, **kwargs):
+        """Set the color used to draw the texts."""
+        if args or kwargs:
+            if len(args) == 1 and args[0] is None:
+                self.opts['textColor'] = None
+            else:
+                self.opts['textColor'] = fn.mkColor(*args, **kwargs)
+        else:
+            self.opts['textColor'] = fn.mkColor(getConfigOption('foreground'))
+
+        self.picture = None
+        self.update()
+
+    def setTickSpacing(self, x=None, y=None):
+        """
+        Set the grid tick spacing to use.
+
+        Tick spacing for each axis shall be specified as an array of
+        descending values, one for each tick scale. When the value
+        is set to None, grid line distance is chosen automatically
+        for this particular level.
+
+        Example:
+            Default setting of 3 scales for each axis:
+            setTickSpacing(x=[None, None, None], y=[None, None, None])
+
+            Single scale with distance of 1.0 for X axis, Two automatic
+            scales for Y axis:
+            setTickSpacing(x=[1.0], y=[None, None])
+
+            Single scale with distance of 1.0 for X axis, Two scales
+            for Y axis, one with spacing of 1.0, other one automatic:
+            setTickSpacing(x=[1.0], y=[1.0, None])
+        """
+        self.opts['tickSpacing'] = (x or self.opts['tickSpacing'][0],
+                                    y or self.opts['tickSpacing'][1])
+
+        self.no_grids = max([len(s) for s in self.opts['tickSpacing']])
```

Comment:
`no_grids` probably stands for 'number of grids'? That might be confusing. Perhaps `griddepth`?